openlcs/packages/views.py

```diff
@@ -876,6 +876,53 @@ def retrieve(self, request, *args, **kwargs):
         return super().retrieve(request, *args, **kwargs)
 
 
+class FilterDuplicateImport(APIView):
+    """
+    Check duplicate container/component/module import,
+    so that we can skip duplicate import step for these files.
+    Duplicate import is container/component/module that exist
+    in the database.
+
+    ####__Request__####
+
+        curl -X POST -H "Content-Type: application/json" \
+-H "Authorization: Token your_token" \
+%(HOST_NAME)s/%(API_PATH)s/components/filter/ \
+-d '{"name": "ubi8-micro-container-source", "version": "", \
+"release": "394.1", "type": "OCI", "license_scan": True, \
+"copyright_scan": True}'
+
+    ####__Response__####
+
+        HTTP 200 OK
+        Content-Type: application/json
+        {
+            "filter_results": {
+                "should_filter": True,
+                "filter_reason": "DB_EXIST"
+            }
+        }
+    """
+    def post(self, request, *args, **kwargs):
+        filter_results = {'should_filter': False}
+        data = request.data
+        imported_components = Component.objects.filter(
+                name=data.get('name', ''),
+                version=data.get('version', ''),
+                release=data.get('release', ''),
+                arch=data.get('arch', ''),
+                type=data.get('type', ''),
+        )
+        serializer = ComponentSerializer(imported_components, many=True)
+        results = serializer.data
+
+        if results and len(results) == 1:
+            filter_results['should_filter'] = True
+            filter_results['filter_reason'] = "DB_EXIST"
```

Comment:
The result could be refined, since this is for checking duplicate, the return should indicate whether it is a duplicate and its duplicate object if it is.

Reply:
ok, will do

Reply:
DONE